src/lib/logical_query_plan/aggregate_node.hpp

```diff
@@ -49,21 +50,21 @@ class AggregateNode : public AbstractLQPNode {
    *
    * NOTE: These functions will possibly result in a full recursive traversal of the ancestors of this node.
    */
-  std::optional<ColumnID> find_column_id_for_expression(const std::shared_ptr<Expression>& expression) const;
-  ColumnID get_column_id_for_expression(const std::shared_ptr<Expression>& expression) const;
+  std::optional<LQPColumnOrigin> find_column_origin_for_expression(
+      const std::shared_ptr<LQPExpression>& expression) const;
+  LQPColumnOrigin get_column_origin_for_expression(const std::shared_ptr<LQPExpression>& expression) const;
   // @}
 
-  std::vector<ColumnID> get_output_column_ids_for_table(const std::string& table_name) const override;
-
   std::string get_verbose_column_name(ColumnID column_id) const override;
 
  protected:
-  std::shared_ptr<AbstractLQPNode> _deep_copy_impl() const override;
+  std::shared_ptr<AbstractLQPNode> _deep_copy_impl(const std::shared_ptr<AbstractLQPNode>& left_child,
+                                                   const std::shared_ptr<AbstractLQPNode>& right_child) const override;
```

Comment:
Docs are in the Declaration in the base class right where they belong - is that not sufficient?

Reply:
Missed it, sorry. But why do I have to give the deep copy method an already copied left_child. Couldn't it copy the child itself?

Reply:
But then every _deep_copy_impl would have to do this - instead of just the base class doing it and passing the generated objects to the impl.

Reply:
Ok, since it is the internal implementation, we can do it that way. I think that renaming the parameters to `copied_left_child` would make it a bit easier to understand.